src/app/mesh/qgsmeshdatasetgrouptreeview.cpp

```diff
@@ -231,55 +228,43 @@ void QgsMeshDatasetGroupTreeView::onSelectionChanged( const QItemSelection &sele
 
   if ( selected.isEmpty() )
   {
-    mActiveGroup = QString();
+    mActiveGroup = 0;
```

Comment:
Shouldn't we use something like -1 for no active group?

And emit activeGroupChanged() signal as well?